### PR TITLE
Update Blocks Tutorial to match Gutenberg Examples

### LINKED ISSUE #20: Consistency between example files and handbook
In the [Gutenberg handbook](https://wordpress.org/gutenberg/handbook/blocks/) is a tutorial "Creating Block Types". This seems to be for very beginners. There is stated:

> To follow along with this tutorial, you can [download the accompanying WordPress plugin](https://github.com/WordPress/gutenberg-examples) which includes all of the examples for you to try on your own site.

The plugin for download is this one with these examples: /[gutenberg-examples](https://github.com/WordPress/gutenberg-examples/tree/5c3dca181f0d5c58a2490ee24b8fbea7af695454)

Comparing source codes you'll find that they are not identical.
For example chapter one "writing your first block type" 

--- enqueuing Block scripts ---
Handbook (hb) `wp_register_script(`
Gutenberg-examples (ge) 01-baisc `wp_enqueue_script(`

(hb) `register_block_type(`
(ge) `blocks.registerBlockType(`

(hb) missing **i18n**

(hb) use of prefix **wp.** `var el = wp.element.createElement`
(ge) `var el = element.createElement;`

(hb) missing `	window.wp.blocks,
	window.wp.i18n,
	window.wp.element`

(hb) missing `filemtime( plugin_dir_path( __FILE__ ) . 'block.js' )`

**And general questions to understand the source code:**

(ge) path = gutenberg-examples/01-basic/block.js
Why the block_type is named **example-01-basic** from `blocks.registerBlockType( 'gutenberg-examples/example-01-basic', {`

What does this  mean? ` step 1 (from the editor).`  from `'Hello World, step 1 (from the editor).'`
The same in (hb) `(Step 1)` from `title: 'Hello World (Step 1)',`

The namings are totally different:
(hb) `    wp_register_script(
        'gutenberg-boilerplate-es5-step01',
        plugins_url( 'step-01/block.js', __FILE__ ),
        array( 'wp-blocks', 'wp-element' )`

(ge) `wp_enqueue_script(
		'gutenberg-examples-01',
		plugins_url( 'block.js', __FILE__ ),
		array( 'wp-blocks', 'wp-i18n', 'wp-element' ),
		filemtime( plugin_dir_path( __FILE__ ) . 'block.js' )`

----------------------------------------
**If you look in the other examples you'll find more inconsistency**

As learning references these examples are not particularly helpful, on the contrary, they are confusing - especially for beginners.

What is wrong, what should I use, why there is a prefix wp. there not, why there it is blocks.register... and there register?... and so on.

### LINKED ISSUE #14005: Documentation: Blocks Tutorial
[Blocks Tutorial](https://wordpress.org/gutenberg/handbook/designers-developers/developers/tutorials/block-tutorial/) links to a repository where one can download a tutorial project to learn basic concepts of Gutenberg Blocks.

Code snippets in the [very next chapter](https://wordpress.org/gutenberg/handbook/designers-developers/developers/tutorials/block-tutorial/writing-your-first-block-type/) is referencing to a different directory structure (can be found [here](https://github.com/ahmadawais/gutenberg-boilerplate-es5)).

While it's not a game-breaker, it's definitely not up to date.

### PR DESCRIPTION
**Pull request by mkaz**
## Description

Updates the block tutorial in Gutenberg Handbook to match the Gutenberg Examples repo.
I cleaned up a few other minor changes, and small fixes.

Fixes #14005 

Fixes https://github.com/WordPress/gutenberg-examples/issues/20

## Types of changes

Documentation.

### CODE REVIEW

File: docs/designers-developers/developers/tutorials/block-tutorial/introducing-attributes-and-editable-fields.md [hunk 1]
```diff
@@ -1,91 +1,125 @@
...
-			type: 'string',
-			source: 'html',
+			type: 'array',
+			source: 'children',
 			selector: 'p',
-		}
+		},
 	},
+```
+
```
Comment (chrisvanpatten) [inline docs/designers-developers/developers/tutorials/block-tutorial/introducing-attributes-and-editable-fields.md]:
```suggestion
```

File: docs/designers-developers/developers/tutorials/block-tutorial/introducing-attributes-and-editable-fields.md [hunk 2]
```diff
@@ -1,91 +1,125 @@
...
+
+When registering a new block type, the `attributes` property describes the shape of the attributes object you'd like to receive in the `edit` and `save` functions. Each value is a [source function](/docs/designers-developers/developers/block-api/block-attributes.md) to find the desired value from the markup of the block.
+
+In the code snippet above, when loading the editor, the `content` value will be extracted from the HTML of the paragraph element in the saved post's markup.
+
+## Components and the `RichText` Component
+
+Earlier examples used the `createElement` function to create DOM nodes, but it's also possible to encapsulate this behavior into "components". This abstraction helps as a pattern to share common behaviors and to hide complexity into self-contained units. There are a number of [components available](/docs/designers-developers/developers/packages/packages-editor.md#components) to use in implementing your blocks. You can see one such component in the code below: the [`RichText` component](/docs/designers-developers/developers/packages/packages-editor.md#richtext).
```
Comment (chrisvanpatten) [inline docs/designers-developers/developers/tutorials/block-tutorial/introducing-attributes-and-editable-fields.md]:
```suggestion
Earlier examples used the `createElement` function to create DOM nodes, but it's also possible to encapsulate this behavior into "components". This abstraction helps you share common behaviors and hide complexity in self-contained units.

There are a number of [components available](/docs/designers-developers/developers/packages/packages-editor.md#components) to use in implementing your blocks. You can see one such component in the code below: the [`RichText` component](/docs/designers-developers/developers/packages/packages-editor.md#richtext).
```

**Review by chrisvanpatten — DISMISSED** [review]
Small tweak but otherwise great. I didn't vet the code too closely, FYI :)

File: docs/designers-developers/developers/tutorials/block-tutorial/writing-your-first-block-type.md [hunk 3]
```diff
@@ -11,18 +11,19 @@ While the block's editor behaviors are implemented in JavaScript, you'll need to
 ```php
 <?php
```
Comment (nosolosw) [inline docs/designers-developers/developers/tutorials/block-tutorial/writing-your-first-block-type.md]: I've noticed that the plugin misses the header so it's shown in the plugins page:

```suggestion
/*
Plugin Name: Gutenberg examples 01
*/
```

File: docs/designers-developers/developers/tutorials/block-tutorial/applying-styles-with-stylesheets.md [hunk 4]
```diff
@@ -1,48 +1,56 @@
...
-In the previous section, the block had applied its own styles by an inline `style` attribute. While this might be adequate for very simple components, you will quickly find that it becomes easier to write your styles by extracting them to a separate stylesheet file.
+In the previous step, the block had applied its own styles by an inline `style` attribute. While this might be adequate for very simple components, you will quickly find that it becomes easier to write your styles by extracting them to a separate stylesheet file.
 
-The editor will automatically generate a class name for each block type to simplify styling. It can be accessed from the object argument passed to the edit and save functions:
+The editor will automatically generate a class name for each block type to simplify styling. It can be accessed from the object argument passed to the edit and save functions. In step 2, we will create a stylesheet to use that class name.
 
 {% codetabs %}
 {% ES5 %}
 ```js
 var el = wp.element.createElement,
```
Comment (nosolosw) [inline docs/designers-developers/developers/tutorials/block-tutorial/applying-styles-with-stylesheets.md]: For consistency with the starting example, we should wrap this within an IIFE passing the blocks and element variables.

File: docs/designers-developers/developers/tutorials/block-tutorial/applying-styles-with-stylesheets.md [hunk 5]
```diff
@@ -51,64 +59,55 @@ registerBlockType( 'gutenberg-boilerplate-esnext/hello-world-step-02', {
...
+
+To enqueue a style that shows on both the front of your site and the editor, use `style` setting.
+
+When registering a block, you can assign one or both of `style` and `editor_style` to respectively assign styles always loaded for a block, or styles only loaded in the editor.
 
-Like scripts, your block's editor-specific styles should be enqueued by assigning the `editor_styles` setting of the registered block type:
+Example 2 shows having a distinct style for each context. Your block is likely to share some styles in both contexts, so in this example you can consider `style.css` as the base stylesheet, placing editor-specific styles in `editor.css`.
 
 ```php
 <?php
```
Comment (nosolosw) [inline docs/designers-developers/developers/tutorials/block-tutorial/applying-styles-with-stylesheets.md]: Same here about plugin header name.

File: docs/designers-developers/developers/tutorials/block-tutorial/applying-styles-with-stylesheets.md [hunk 6]
```diff
@@ -51,64 +59,55 @@ registerBlockType( 'gutenberg-boilerplate-esnext/hello-world-step-02', {
 The class name is generated using the block's name prefixed with `wp-block-`, replacing the `/` namespace separator with a single `-`.
 
 ```css
```
Comment (nosolosw) [inline docs/designers-developers/developers/tutorials/block-tutorial/applying-styles-with-stylesheets.md]: If this is `editor.css`, can we perhaps mention that by passing and also say that `style.css` will be the same but with different colors?

File: docs/designers-developers/developers/tutorials/block-tutorial/applying-styles-with-stylesheets.md (hunk 4 above)
Comment (mkaz) [reply docs/designers-developers/developers/tutorials/block-tutorial/applying-styles-with-stylesheets.md]: :+1:

File: docs/designers-developers/developers/tutorials/block-tutorial/applying-styles-with-stylesheets.md (hunk 5 above)
Comment (mkaz) [reply docs/designers-developers/developers/tutorials/block-tutorial/applying-styles-with-stylesheets.md]: :+1:

File: docs/designers-developers/developers/tutorials/block-tutorial/applying-styles-with-stylesheets.md (hunk 6 above)
Comment (mkaz) [reply docs/designers-developers/developers/tutorials/block-tutorial/applying-styles-with-stylesheets.md]: :+1:

**Comment by mkaz** [comment]
@nosolosw all the suggestions were implemented, thanks!

**Review by oandregal — APPROVED** [review]
I've pushed some fixes and all examples are now fully functioning. Let's have this :ship:

**Comment by gziolo** [comment]
Great work, for better or worse we have both `gutenberg-examples` and this tutorial aligned. However,  I noticed that in some places we could further improve `gutenberg-examples`. I think that would be great to focus on it on one of the Contributor Days at WordCamps :)

**Comment by nosolosw** [comment]
@chrisvanpatten have your changes being addressed? I think we can't merge until you either clear the "requested changes" status or give a :+1: (at least I don't see the option).

**Pull request merged**